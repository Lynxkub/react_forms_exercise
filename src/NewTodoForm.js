import React , { useState } from 'react';



const NewTodoForm = ({addTodo}) => {
   
    const INITIAL_STATE = {
        todo: ""
    }

    const [formData , setFormData] = useState(INITIAL_STATE);
    
    const handleChange = (e) => {
        const {name  , value } = e.target;

        setFormData(formData => ({
            ...formData,
            [name] : value
        }))
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        addTodo({...formData});
        setFormData(INITIAL_STATE);
    }


    return (
        <div>
        <form onSubmit = {handleSubmit}>
            <label htmlFor = 'todo'>New Todo</label>
            <input 
            data-testid = 'todo'
            id = 'todo'
            name = 'todo'
            placeholder = 'New Todo'
            type = 'text'
            value = {formData.todo}
            onChange={handleChange}
            />
            <button>Add Todo</button>
        </form>
        </div>
    )
}


export default NewTodoForm;