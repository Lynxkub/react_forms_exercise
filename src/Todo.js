import React from 'react';


const Todo = ({todo , id , deleteTodo}) =>  {

    const handleClick = () => {
        deleteTodo(id);
    }
    return (
        <div>
            <div style = {{display : 'inline-block'}}>{todo}</div>
            <button onClick = {handleClick}>Remove Todo</button>
        </div>
    )
}

export default Todo;