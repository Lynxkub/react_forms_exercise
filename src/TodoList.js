import React , { useState } from 'react';
import uuid from 'react-uuid';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo'



const TodoList = () => {

    const [todos , setTodos] = useState([]);

   const addTodo = (newTodo) => {
       setTodos(todos => [...todos , {...newTodo , id : uuid() }])
   }

   const deleteTodo = (id) => {
       setTodos(todos => todos.filter(todo => todo.id !== id))
   }

    return (
        <div>
            <h3>Todo List</h3>
            <NewTodoForm  addTodo={addTodo} />
            <div>
                {todos.map(todo => <Todo id={todo.id} key = {todo.id} todo = {todo.todo} deleteTodo = {deleteTodo}/>)}
            </div>
        </div>
    )
}


export default TodoList;