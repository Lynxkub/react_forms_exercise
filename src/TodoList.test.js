import React from 'react';
import {render , fireEvent} from '@testing-library/react'
import Todo from './Todo';
import TodoList from './TodoList';
import NewTodoForm from './NewTodoForm';
import '@testing-library/jest-dom/extend-expect';


it('renders without crashing' , function () {
    render(<TodoList />)
})


it('matches snapshot' , function () {
    const {asFragment} = render(<TodoList />)
    expect(asFragment()).toMatchSnapshot();
})

it('created and deletes todos' , function () {
    const {queryByText , getByTestId} = render(<TodoList />);
    const todo = getByTestId('todo');
    const btn = queryByText('Add Todo');
    // Before todo is created
    expect(queryByText('Remove Todo')).not.toBeInTheDocument();
    // Creating a todo 
    fireEvent.change(todo , {target : {value : 'mow lawn'}});
    fireEvent.click(btn);
    expect(queryByText('Remove Todo')).toBeInTheDocument();
    // Deleting the todo
    fireEvent.click(queryByText('Remove Todo'));
    expect(queryByText('Remove Todo')).not.toBeInTheDocument();
})