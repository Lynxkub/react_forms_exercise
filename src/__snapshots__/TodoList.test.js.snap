// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Todo List
    </h3>
    <div>
      <form>
        <label
          for="todo"
        >
          New Todo
        </label>
        <input
          data-testid="todo"
          id="todo"
          name="todo"
          placeholder="New Todo"
          type="text"
          value=""
        />
        <button>
          Add Todo
        </button>
      </form>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
